src/api-service/__app__/onefuzzlib/agent_authorization.py

```diff
@@ -54,6 +55,8 @@ def try_get_token_auth_header(request: func.HttpRequest) -> Union[Error, TokenDa
 
 @cached(ttl=60)
 def is_authorized(token_data: TokenData) -> bool:
+    if get_scaleset_principal_id() == token_data.object_id:
```

Comment:
Let's add a comment describing what situation this is checking, at a higher level. Is this an optimization? Or does the check below miss something?